src/ui/reducers/layout.ts

```diff
@@ -0,0 +1,26 @@
+import { UIState } from "ui/state";
+import { LayoutState } from "ui/state/layout";
+import { LayoutAction } from "ui/actions/layout";
+
+function initialLayoutState(): LayoutState {
+  return {
+    showCommandPalette: false,
+  };
+}
+
+export default function update(state = initialLayoutState(), action: LayoutAction): LayoutState {
+  switch (action.type) {
+    case "set_show_command_palette": {
+      return {
+        ...state,
+        showCommandPalette: action.value,
+      };
+    }
+
+    default: {
+      return state;
+    }
+  }
+}
+
+export const getShowCommandPalette = (state: UIState) => state.layout.showCommandPalette;
```

Comment:
It's a shame, this is *almost* simple enough that it doesn't belong in redux, but it's probably right to have it here.